crates/project_panel/src/project_panel.rs

```diff
@@ -3621,6 +3639,140 @@ mod tests {
+        panel.update(cx, |panel, cx| {
+            panel.copy(&Default::default(), cx);
+        });
+
+        select_path(&panel, "root2/one.txt", cx);
+        panel.update(cx, |panel, cx| {
+            panel.select_next(&Default::default(), cx);
+            panel.paste(&Default::default(), cx);
+        });
+        cx.executor().run_until_parked();
+        assert_eq!(
+            visible_entries_as_strings(&panel, 0..50, cx),
+            &[
+                //
+                "v root1",
+                "    > a",
+                "      one.txt",
+                "      three.txt",
+                "      two.txt",
+                "v root2",
+                "    > b",
+                "      four.txt",
+                "      one.txt",
+                "      three.txt",
+                "      two.txt  <== selected",
+            ]
+        );
+
+        select_path(&panel, "root1/three.txt", cx);
+        panel.update(cx, |panel, cx| {
+            panel.copy(&Default::default(), cx);
+        });
+        select_path(&panel, "root2/two.txt", cx);
+        panel.update(cx, |panel, cx| {
+            panel.select_next(&Default::default(), cx);
+            panel.paste(&Default::default(), cx);
+        });
+        // TODO, paste same file in worktree, will emit prompt for select replace or cancel
+        // current test, default cancel same file paste different worktree
+
+        cx.executor().run_until_parked();
+        assert_eq!(
+            visible_entries_as_strings(&panel, 0..50, cx),
+            &[
+                //
+                "v root1",
+                "    > a",
+                "      one.txt",
+                "      three.txt",
+                "      two.txt",
+                "v root2",
+                "    > b",
+                "      four.txt",
+                "      one.txt",
+                "      three copy.txt",
+                "      three.txt",
+                "      two.txt  <== selected",
+            ]
+        );
+
+        select_path(&panel, "root1/a", cx);
+        panel.update(cx, |panel, cx| {
+            panel.copy(&Default::default(), cx);
+        });
+        select_path(&panel, "root2/two.txt", cx);
+        panel.update(cx, |panel, cx| {
+            panel.select_next(&Default::default(), cx);
+            panel.paste(&Default::default(), cx);
+        });
+
+        cx.executor().run_until_parked();
+        assert_eq!(
+            visible_entries_as_strings(&panel, 0..50, cx),
+            &[
+                //
+                "v root1",
+                "    > a",
+                "      one.txt",
+                "      three.txt",
+                "      two.txt",
+                "v root2",
+                "    > a",
+                "    > b",
+                "      four.txt",
+                "      one.txt",
+                "      three copy.txt",
+                "      three.txt",
+                "      two.txt  <== selected",
```

Comment:
Same note about the selection, check out how pasted dirs are selected and ensure we do the same.